neighsyncd/restore_neighbors.py

```diff
@@ -33,8 +33,9 @@
 # timeout the restore process in 1 min if not finished
 # This is mostly to wait for interfaces to be created and up after system warm-reboot
 # and this process is started by supervisord in swss docker.
-# It would be good to keep that time below routing reconciliation time-out.
-TIME_OUT = 60
+# It would be good to keep that time consistent routing reconciliation time-out, here
+# we are upstream, we shouldn't give up before down stream.
+TIME_OUT = 120
```

Comment:
If neighbor restore should run when all interfaces becomeready can we subscribe to state db and listen untill all interfaces become ready instead of relying on the timer?

Reply:
@zhenggen-xu can you chime in?

Reply:
The code check the interfaces state in the kernel before we put neighbors onto the interface, there was a check interval, so not relying on timer if interface are in a right state.  It would be better to check the kernel state instead of statedb since we need to make sure IP addresses are configured and operationally up, which do not exist in stateDB today.

Anyway, the timer is required whether we are using polling mode or subscribe mode.   This is to make sure we don't stuck there forever, e,g, the port never come up due to physical down.

Reply:
I thought waiting for *all* interfaces could delay restoring some of the entries. Do we have the consideration of restore as soon as they ready thoughts here too?

Reply:
The restoring process was done per interface state.  But we only have one signal to tell neighsyncd that the restore is done. To have per interface signal would complicate the solution.

Reply:
I didn't meant to ask for per interface signal. I was saying that current approach is trying best to bring back neighbors as interfaces individually getting ready.

Reply:
right, that is the current behavior.